### PR TITLE
Update docs about dockershim removal

### PR DESCRIPTION
**Pull request by twz123**
## Description

The official docs about the "metrics situation" have been merged. Update the links accordingly. Also do some slight spelling and case unifications, and add a nav link to the dockershim docs page.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update

## How Has This Been Tested?

- [ ] Manual test
- [ ] Auto test added

## Checklist:

- [x] My code follows the style [guidelines](https://github.com/k0sproject/k0s/blob/main/docs/contributors/overview.md) of this project 
- [x] My commit messages are [signed-off](https://github.com/k0sproject/k0s/blob/main/docs/contributors/github_workflow.md)
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Review by juanluisvaladas — CHANGES_REQUESTED** [review]
Other than the sentence, LGTM

File: docs/troubleshooting.md [hunk 1]
```diff
@@ -144,11 +144,27 @@ LD_FLAGS="--custom-flag=value" make k0s
...
-In the future kubelet will be refactored to get the container metrics from CRI interface rather than from the runtime directly. This work is specified and followed up in [KEP-2371](https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/2371-cri-pod-container-stats/README.md) but until that work completes the only option is to run a standalone cAdvisor. There's ongoing [effort](https://github.com/kubernetes/website/issues/30681) to both document the current shortcomings and how to run standalone cAdvisor in Kubernetes community.
+Due to removal of the embedded dockershim from Kubelet, the Kubelet's embedded
+[cAdvisor] metrics got slightly broken. If your container runtime is a custom
+containerd you can add
+`--kubelet-extra-flags="--containerd=<path/to/containerd.sock>"` into k0s worker
+startup. That configures the Kubelet embedded cAdvisor to talk directly with
+containerd to gather the metrics and thus gets the expected labels in place.
+
+Unfortunately this does not work on when using Docker via cri-dockerd shim.
+There's currently no easy workarounds for this.
```
[the remark this answers is not in the document]
Comment (twz123) [reply docs/troubleshooting.md:155]: What about

> Currently, there is no easy solution to this problem.

?

[the remark this answers is not in the document]
Comment (juanluisvaladas) [reply docs/troubleshooting.md:155]: I like that

[the remark this answers is not in the document]
Comment (juanluisvaladas) [reply docs/troubleshooting.md:155]: I like that.

[the remark this answers is not in the document]
Comment (twz123) [reply docs/troubleshooting.md]: Done.

**Review by juanluisvaladas — APPROVED** [review]
LGTM

**Pull request merged**